### PR TITLE
App testing

### PR DESCRIPTION
**Pull request by JamesRexMiller4**
#### What's this PR do?

-This PR creates a Loading component and gives it functionality/css animation for when a user is awaiting the movies fetching to resolve

-This PR also writes tests for the Loading, App, and MoviePage components, tests for the apiCalls, and the movies reducer. 

#### Where should the reviewer start?

Reviewer should probably start with the testing files to make sure they are robust. I would start with the apiCalls and movies reducer as those are fairly straight forward. Next I would look at the App test and make sure that everything that should be tested is covered. Finally, I would review the MoviePage testing and make sure that everything that needs to be tested is covered. 

Note: There are a number of snapshots for the MoviePage and as the findMovieRating method is conditionally rendering html based on the value of its inputs, I opted to let the snapshots account for the testing of that method. 

#### How should this be manually tested?

Npm test or you can run npm start and try refreshing your browser a number of times to try and see the Loading component in action.

### CODE REVIEW

File: rancid-tomatillos/src/Components/Loading/Loading.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  height: 25vw;
+  background: {
+    image: url('../../images/popcorn.png');
+    size: cover;
+    repeat: no-repeat;
+  }
+  animation: pulse 2s infinite;
+}
+
+@keyframes pulse {
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/Components/Loading/Loading.scss:25]: Nice work implementing @keyframes to animate the loading component! Looks really good!

File: rancid-tomatillos/src/apiCalls.test.js [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+      window.fetch = jest.fn().mockImplementation(() => {
+        return Promise.resolve({
+          ok: false
+        })
+      });
+
+      expect(postUserRating(mockOptions, mockUserId)).rejects.toEqual(Error('Failure to POST User Rating'));
+    });
+    
+    it('SAD: should throw an error message if promise does not resolve', () => {
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/apiCalls.test.js:228]: Awesome work testing sad and happy paths on the apicalls! This looks like some robust testing!

File: rancid-tomatillos/src/containers/MoviePage/MoviePage.test.js [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+    it('should match the snapshot when the isLoggedIn is false', () => {
+      const isLoggedIn = false
+      wrapper = shallow(<MoviePage 
+        id={15}
+        movies={movies}
+        isLoggedIn={isLoggedIn}
+        userId={userId}
+        userRatings={userRatings}
+      />)
+      expect(wrapper).toMatchSnapshot();
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/containers/MoviePage/MoviePage.test.js:121]: awesome work testing all of the conditional rendering via snapshots!!

**Review by adamsjr8576 — APPROVED** [review]
This all looks great! Really nice work writing those tests our to be really robust and test all potential paths. The loading component looks really good too!

File: rancid-tomatillos/public/index.html [hunk 4]
```diff
@@ -24,7 +24,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React App</title>
+    <title>Rancid Tomatillos</title>
```
Comment (ckoga) [inline rancid-tomatillos/public/index.html:27]: Nice catch!

**Review by ckoga — APPROVED** [review]

**Pull request merged**